builder/azure/arm/step_deploy_template.go

```diff
@@ -335,6 +350,8 @@ func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, subs
 					s.say(fmt.Sprintf("Couldn't delete %s resource. Will retry.\n"+
```

Comment:
This should probably become a `log.Printf` so we don't flood users with non-actionable error messages
```suggestion
					log.Printf("[INFO] Couldn't delete resource %s.%s, will retry", resourceType, resourceName)
```

Reply:
That's fair, we don't expect this error to occur anymore unless something is actually going weird though, but since we're gonna retry it seems worthwhile to add that into logs instead.